flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/schema/PulsarSchemaUtils.java

```diff
@@ -0,0 +1,183 @@
+ */
+
+package org.apache.flink.connector.pulsar.common.schema;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.TypeExtractionUtils;
+import org.apache.flink.connector.pulsar.common.schema.factories.BooleanSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.ByteBufSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.ByteBufferSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.ByteSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.BytesSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.DateSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.DoubleSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.FloatSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.InstantSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.IntSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.KeyValueSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.LocalDateSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.LocalDateTimeSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.LocalTimeSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.LongSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.NoneSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.ShortSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.StringSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.TimeSchemaFactory;
+import org.apache.flink.connector.pulsar.common.schema.factories.TimestampSchemaFactory;
+import org.apache.flink.util.InstantiationUtil;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.common.schema.SchemaInfo;
+import org.apache.pulsar.common.schema.SchemaType;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static java.util.Collections.emptySet;
+
+/**
+ * Util class for pulsar schema. Register all the {@link PulsarSchemaFactory} in this class and
+ * provide the {@link TypeInformation} or {@link SerializablePulsarSchema} conversion.
+ */
+@Internal
+public final class PulsarSchemaUtils {
+
+    private static final ConcurrentMap<SchemaType, Set<PulsarSchemaFactory<?>>> FACTORY_REGISTER =
+            new ConcurrentHashMap<>();
+
+    static {
+        // Register the predefined schema factories here.
+        registerSchemaFactory(ByteSchemaFactory.class);
```

Comment:
Instead of registering a class and instantiate it with refactoring, I'd pass an instance here.
`registerSchemaFactory(new ByteSchemaFactory());`
That approach allows us to replace it with
`registerSchemaFactory(new PrimitiveSchemaFactory(ByteSchema.of(), Types.BYTE));` with the implementation outlined later.